solr/core/src/java/org/apache/solr/handler/component/QueryElevationComponent.java

```diff
@@ -509,7 +509,9 @@ private void setQuery(ResponseBuilder rb, Elevation elevation) {
       rb.setQuery(new BoostQuery(elevation.includeQuery, 0f));
     } else {
       BooleanQuery.Builder queryBuilder = new BooleanQuery.Builder();
-      queryBuilder.add(rb.getQuery(), BooleanClause.Occur.SHOULD);
+      BooleanClause.Occur queryOccurrence = params.getBool(QueryElevationParams.ELEVATE_ONLY_DOCS_MATCHING_QUERY, false) ?
+              BooleanClause.Occur.SHOULD : BooleanClause.Occur.MUST;
```

Comment:
@tkaessmann when I ran tests I saw this line has a bug. It inverts the SHOULD and MUST.
Hopefully I fixed it and merged it without the bug.